ui/app/src/components/shared/ResourceCard.tsx

```diff
@@ -35,6 +37,29 @@ export const ResourceCard: React.FunctionComponent<ResourceCardProps> = (props:
   );
   const navigate = useNavigate();
 
+  let costsTagsRoles: string[] = []
+
+  switch(props.resource.resourceType) {
```

Comment:
isn't this switch duplicating what's happening within the useCallBack code below?

i.e. couldn't we have one switch that sets variables such as `costsTagsRoles` and `resourceUrl`, and then use those variables in the code below?

Reply:
Check the changes.